internal/clients/builder.go

```diff
@@ -167,8 +168,11 @@ func Build(ctx context.Context, builder ClientBuilder) (*Client, error) {
 	if features.EnhancedValidationEnabled() {
 		subscriptionId := commonids.NewSubscriptionID(client.Account.SubscriptionId)
 
-		location.CacheSupportedLocations(ctx, *resourceManagerEndpoint)
-		if err := resourceproviders.CacheSupportedProviders(ctx, client.Resource.ResourceProvidersClient, subscriptionId); err != nil {
+		ctx2, cancel := context.WithTimeout(ctx, 10*time.Minute)
```

Comment:
an aside: we should probably think about surfacing the available Locations on the Environment, so that we don't need to hit this endpoint twice?

Reply:
good idea, will have a think about it 👍